build.fsx

```diff
@@ -36,14 +36,14 @@ let parsedRelease =
     File.ReadLines "RELEASE_NOTES.md"
     |> ReleaseNotesHelper.parseReleaseNotes
 
-//Fake.ReleaseNotesHelper.Parse assumes letters+int in PreRelease.TryParse. 
-//This means we cannot append the full date yyyMMddHHmmss to prerelease. 
-//See https://github.com/fsharp/FAKE/issues/522
-//TODO: When this has been fixed, switch to DateTime.UtcNow.ToString("yyyyMMddHHmmss")
-let preReleaseVersion = parsedRelease.AssemblyVersion + "-" + (getBuildParamOrDefault "nugetprerelease" "pre") + DateTime.UtcNow.ToString("yyMMddHHmm")
+let envBuildNumber = System.Environment.GetEnvironmentVariable("BUILD_NUMBER")
+let buildNumber = if String.IsNullOrWhiteSpace(envBuildNumber) then "0" else envBuildNumber
+
+let version = parsedRelease.AssemblyVersion + "." + buildNumber
+
 let isUnstableDocs = hasBuildParam "unstable"
 let isPreRelease = hasBuildParam "nugetprerelease"
-let release = if isPreRelease then ReleaseNotesHelper.ReleaseNotes.New(parsedRelease.AssemblyVersion, preReleaseVersion, parsedRelease.Notes) else parsedRelease
+let release = if isPreRelease then ReleaseNotesHelper.ReleaseNotes.New(version, version + " -beta", parsedRelease.Notes) else ReleaseNotesHelper.ReleaseNotes.New(version, version, parsedRelease.Notes) 
```

Comment:
Nah, because the build number for release builds doesn't really reflect what's going on - it increments regardless of changes in major / minor / revision number.